src/frontend/src/components/public/PIAFormTabs/review/index.tsx

```diff
@@ -113,7 +112,22 @@ const PIAReview = ({ printPreview }: IReviewProps) => {
       true,
     );
   };
+  const allowUserReviewProgramArea = () => {
+    const userGuid = getGUID();
+    const selectedRoles = reviewForm?.programArea?.selectedRoles;
+    // if selectedRoles is null means none of selectedRole got reviewed so return true
+    // otherwise loop all the role in reviews part to see if the current user already did review
+    if (selectedRoles === null) return true;
+    for (const role of selectedRoles) {
```

Comment:
should this be done on a case by case bases? this function is called to loop and if even one matches progamareaguiluserguid the whole review page turns to edit mode? Not very clear